subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/codenarc/CodeNarcPluginToolchainsIntegrationTest.groovy

```diff
@@ -32,6 +33,7 @@ import org.junit.Assume
  * {@code CodeNarc} tasks behave as expected.
  */
 @TargetCoverage({ CodeNarcCoverage.ALL })
+@Requires(adhoc = { AvailableJavaHomes.differentVersion })
```

Comment:
Given the conversations, this is not a satisfying solution. Either pull that change from this PR to not delay it further, or apply one of the suggestions.